lib/elix/responders/lists.ex

```diff
@@ -0,0 +1,105 @@
+defmodule Elix.Responders.Lists do
+  @moduledoc """
+  Commands to create, update, and delete one-dimensional lists and their items.
+  """
+
+  use Hedwig.Responder
+  alias Hedwig.Message
+  alias Elix.Lists
+
+  @usage """
+  show lists - Displays all lists
+  """
+  respond ~r/show lists\Z/i, msg do
+    reply(msg, render_items(Lists.all))
+  end
+
+  @usage """
+  create list <name> - Creates a new list with name
+  """
+  respond ~r/create list (.+)/i, %Message{matches: %{1 => list_name}} = msg do
+    Lists.create(list_name)
+    reply(msg, render_items(Lists.all))
+  end
+
+  @usage """
+  show list <list> - Displays the contents of a list by name or number
+  """
+  respond ~r/show list (.+)/i, %Message{matches: %{1 => list_id}} = msg do
+    list_name = parse_list_identifier(list_id)
```

Comment:
I'm wondering if I can offload some of the work of `parse_list_identifier` into the regex itself. With named captures, I *think* I can have the regex key the match as `number` if it's a string of digits, otherwise as `name`. Here's a very naive example that just matches strings of digits vs. strings of non-digits:

```elixir
iex> Regex.named_captures(~r/test (?<name>\D+)|(?<number>\d+)\Z/, "Gonna test 1234")
%{"name" => "", "number" => "1234"}

iex> Regex.named_captures(~r/test (?<name>\D+)|(?<number>\d+)\Z/, "Gonna test this thing")
%{"name" => "this thing", "number" => ""}
```

Of course, that breaks down with mixed strings:

```elixir
iex> Regex.named_captures(~r/test (?<name>\D+\Z)|(?<number>\d+\Z)/, "Gonna test this thing 9 times")
nil
```

but the first named capture is naive, matching "strings composed entirely of non-digits" as `name`. What I want to match are "strings not entirely composed of digits". Now, to figure that out at 6am...